### PR TITLE
Document how to configure XContent parsing constraints

### PR DESCRIPTION
**Pull request by reta**
### Description
Document how to configure XContent parsing constraints, targets 2.8.0 release

### Issues Resolved
Closes https://github.com/opensearch-project/documentation-website/issues/4089

### Checklist
- [X] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by reta** [comment]
@Naarcha-AWS the `2.8.0` is mostly there could we please get it in before the release? thank you!

**Comment by hdhalter** [comment]
Thanks, @reta , @Naarcha-AWS is OOO this week. 

@kolchfa-aws , @cwillum , or @vagimeli - can one of you please take a look at this? Thanks.

File: _install-and-configure/install-opensearch/index.md [hunk 1]
```diff
@@ -90,3 +90,11 @@ The [sample docker-compose.yml]({{site.url}}{{site.baseurl}}/install-and-configu
 Do not declare the same JVM options in multiple locations because it can result in unexpected behavior or a failure of the OpenSearch service to start. If you declare JVM options using an environment variable, such as `OPENSEARCH_JAVA_OPTS=-Xms3g -Xmx3g`, then you should comment out any references to that JVM option in `config/jvm.options`. Conversely, if you define JVM options in `config/jvm.options`, then you should not define those JVM options using environment variables.
 {: .note}
 
+## Important system properties
+
+The OpenSearch has a number of system properties that you could specify in `config/jvm.options` or `OPENSEARCH_JAVA_OPTS`, using `-D` command line arguments notation.
```
Comment (kolchfa-aws) [inline _install-and-configure/install-opensearch/index.md:95]:
```suggestion
The OpenSearch has a number of system properties that you can specify in `config/jvm.options` or `OPENSEARCH_JAVA_OPTS` using `-D` command line argument notation.
```

File: _install-and-configure/install-opensearch/index.md [hunk 2]
```diff
@@ -90,3 +90,11 @@ The [sample docker-compose.yml]({{site.url}}{{site.baseurl}}/install-and-configu
...
+
+- `opensearch.xcontent.string.length.max=<value>`
```
Comment (kolchfa-aws) [inline _install-and-configure/install-opensearch/index.md:97]: Could we add an introductory sentence for this snippet? How about this:
````suggestion
For example, the following setting sets the `opensearch.xcontent.string.length.max` to `<value>`:

```plaintext
- opensearch.xcontent.string.length.max=<value>
```
````

File: _install-and-configure/install-opensearch/index.md [hunk 3]
```diff
@@ -90,3 +90,11 @@ The [sample docker-compose.yml]({{site.url}}{{site.baseurl}}/install-and-configu
...
+
+  By default, OpenSearch does not impose any limits on the maximum length of the JSON string fields. To protect your clusters from pontential DDoS or memory issues, you could set the system property `opensearch.xcontent.string.length.max` to the reasonable boundary (the maximum is `2147483647`), for example: `-Dopensearch.xcontent.string.length.max=5000000`.
```
Comment (kolchfa-aws) [inline _install-and-configure/install-opensearch/index.md:99]:
```suggestion
By default, OpenSearch does not impose any limits on the maximum length of the JSON string fields. To protect your cluster from potential distributed denial-of-service (DDoS) or memory issues, you can set the `opensearch.xcontent.string.length.max` system property to a reasonable limit (the maximum is 2,147,483,647), for example `-Dopensearch.xcontent.string.length.max=5000000`.
```

**Review by kolchfa-aws — APPROVED** [review]
LGTM with comments. Thanks, @reta!

File: _install-and-configure/install-opensearch/index.md (hunk 2 above)
Comment (reta) [reply _install-and-configure/install-opensearch/index.md:97]: My intention here was to keep it as a lit, similarly to the previous section (not changed here) with system parameters.

File: _install-and-configure/install-opensearch/index.md [hunk 4]
```diff
@@ -90,3 +90,11 @@ The [sample docker-compose.yml]({{site.url}}{{site.baseurl}}/install-and-configu
 Do not declare the same JVM options in multiple locations because it can result in unexpected behavior or a failure of the OpenSearch service to start. If you declare JVM options using an environment variable, such as `OPENSEARCH_JAVA_OPTS=-Xms3g -Xmx3g`, then you should comment out any references to that JVM option in `config/jvm.options`. Conversely, if you define JVM options in `config/jvm.options`, then you should not define those JVM options using environment variables.
 {: .note}
 
+## Important system properties
+
+The OpenSearch has a number of system properties that you can specify in `config/jvm.options` or `OPENSEARCH_JAVA_OPTS` using `-D` command line argument notation.
```
Comment (kolchfa-aws) [inline _install-and-configure/install-opensearch/index.md:95]: OK sure. Then how about this
```suggestion
The OpenSearch has a number of system properties that you can specify in `config/jvm.options` or `OPENSEARCH_JAVA_OPTS` using `-D` command line argument notation, for example:
```

File: _install-and-configure/install-opensearch/index.md (hunk 2 above)
Comment (kolchfa-aws) [reply _install-and-configure/install-opensearch/index.md:97]: Understood. See my suggestion above.

File: _install-and-configure/install-opensearch/index.md [hunk 5]
```diff
@@ -90,3 +90,11 @@ The [sample docker-compose.yml]({{site.url}}{{site.baseurl}}/install-and-configu
 Do not declare the same JVM options in multiple locations because it can result in unexpected behavior or a failure of the OpenSearch service to start. If you declare JVM options using an environment variable, such as `OPENSEARCH_JAVA_OPTS=-Xms3g -Xmx3g`, then you should comment out any references to that JVM option in `config/jvm.options`. Conversely, if you define JVM options in `config/jvm.options`, then you should not define those JVM options using environment variables.
 {: .note}
 
+## Important system properties
+
+The OpenSearch has a number of system properties that you can specify in `config/jvm.options` or `OPENSEARCH_JAVA_OPTS` using `-D` command line argument notation, for example:
+
+- `opensearch.xcontent.string.length.max=<value>`
+
+By default, OpenSearch does not impose any limits on the maximum length of the JSON string fields. To protect your cluster from potential distributed denial-of-service (DDoS) or memory issues, you can set the `opensearch.xcontent.string.length.max` system property to a reasonable limit (the maximum is 2,147,483,647), for example `-Dopensearch.xcontent.string.length.max=5000000`.
```
Comment (natebower) [inline _install-and-configure/install-opensearch/index.md:99]:
```suggestion
By default, OpenSearch does not impose any limits on the maximum length of the JSON string fields. To protect your cluster from potential distributed denial-of-service (DDoS) or memory issues, you can set the `opensearch.xcontent.string.length.max` system property to a reasonable limit (the maximum is 2,147,483,647), for example, `-Dopensearch.xcontent.string.length.max=5000000`.
```

**Review by natebower — APPROVED** [review]
@kolchfa-aws Approved with noted changes. Thanks!

**Pull request merged**